### PR TITLE
[Impeller] Adjustments to SubmitKHR and queries.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Removes the rest of the validation errors by increasing the size of the query ring buffer. Because we rely on callbacks fired from the fence waiter, when the phone is under load these can be substantially delayed from the frame finishing.

Adjusts presentation to use a dedicated default priority/slow core affinity/single threaded task runner. From the ARM video guide, submitKHR will block until all previously submitted work has been scheduled. This needs to happen in order, and doesn't need to run at high priority or even on medium speed cores.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
I can't provide it, but I think this fixes https://github.com/flutter/flutter/issues/135642

File: impeller/renderer/backend/vulkan/context_vk.cc [hunk 1]
```diff
@@ -128,6 +129,12 @@ void ContextVK::Setup(Settings settings) {
     return;
   }
 
+  submit_message_loop_ = fml::ConcurrentMessageLoop::Create(1u);
+  submit_message_loop_->PostTaskToAllWorkers([]() {
+    // submitKHR is extremely cheap and mostly blocks on an internal fence.
+    fml::RequestAffinity(fml::CpuAffinity::kEfficiency);
```
Comment (chinmaygarde) [inline impeller/renderer/backend/vulkan/context_vk.cc:135]: How about just fml::Thread? That has a handy way to name the thread too. You can still post task to it to set it as an efficiency affinity.

File: impeller/renderer/backend/vulkan/context_vk.h [hunk 2]
```diff
@@ -176,6 +188,7 @@ class ContextVK final : public Context,
   std::shared_ptr<CommandPoolRecyclerVK> command_pool_recycler_;
   std::string device_name_;
   std::shared_ptr<fml::ConcurrentMessageLoop> raster_message_loop_;
+  std::shared_ptr<fml::ConcurrentMessageLoop> submit_message_loop_;
```
Comment (chinmaygarde) [inline impeller/renderer/backend/vulkan/context_vk.h:191]: Here and elsewhere, perhaps call it queue_submit_task_runner? I was initially wondering, "submit what"?

**Review by chinmaygarde — APPROVED** [review]
Suggestion about using fml::Thread instead of a "concurrent" runner with count 1. But just nits. LGTM.

File: impeller/renderer/backend/vulkan/context_vk.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/vulkan/context_vk.cc:135]: Oh neat, didn't realize that. makes more sense

File: impeller/renderer/backend/vulkan/context_vk.h (hunk 2 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/vulkan/context_vk.h]: Done

File: impeller/renderer/backend/vulkan/context_vk.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/vulkan/context_vk.cc:135]: Done

**Review by chinmaygarde — APPROVED** [review]

File: impeller/renderer/backend/vulkan/context_vk.cc [hunk 3]
```diff
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 #include "impeller/renderer/backend/vulkan/context_vk.h"
+#include "fml/concurrent_message_loop.h"
```
Comment (chinmaygarde) [inline impeller/renderer/backend/vulkan/context_vk.cc:6]: Nit: Newline between primary header and the rest.

Comment (jonahwilliams) [reply impeller/renderer/backend/vulkan/context_vk.cc:7]: Done

**Pull request merged**